expr: Implemented __trunc__ method

abaokar-07 opened the pull request:
Added the __trunc__ method as the one in math.trunc method of the math module

@asmeurer please review

abaokar-07 committed:
Update expr.py

asmeurer commented on sympy/core/expr.py:
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
 
+    def __trunc__(self):
+        r = self.round(2)
+        if self in(S.Nan, S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
```
Put a space after `in`

asmeurer commented on sympy/core/expr.py:
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
         return StrictLessThan(self, other, evaluate=False)
 
+    def __trunc__(self):
+        r = self.round(2)
```
Why not just use `int(self)` here?

asmeurer commented on sympy/core/expr.py:
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
+        if not r.is_Number:
+            raise TypeError("can't truncate symbols and expressions")
+        else:
+            return int(r)
```
This should return a SymPy type.

asmeurer commented:
Please add tests.

abaokar-07 committed:
modified expr.py

abaokar-07 commented:
@asmeurer I have modified `expr.py` according to your reviews
But before adding tests, I wanted to ask one thing that after I modified expr, I gave input as `trunc(3.67)` and I got the following error : `TypeError: trunc() missing 1 required positional argument: 'p'`

asmeurer commented:
You need to use `math.trunc`. The `trunc` that comes with SymPy is unrelated.

abaokar-07 committed:
Added tests and more modifications to expr

abaokar-07 commented:
@asmeurer Added the tests

abaokar-07 commented:
@asmeurer what's the status on this, any more changes are needed or is it fine?

asmeurer commented on sympy/core/expr.py:
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
+        if not self.is_number:
+            raise TypeError("can't truncate symbols and expressions")
+        r = int(self)
+        if r in (S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
```
r cannot be one of these.

abaokar-07 committed:
modified expr

abaokar-07 commented:
@asmeurer are the latest changes fine?

abaokar-07 commented:
@asmeurer is the modified form okay?

abaokar-07 commented:
@asmeurer ping please, is the change good?

asmeurer commented on sympy/core/expr.py:
```diff
@@ -344,6 +344,14 @@ def __lt__(self, other):
+    def __trunc__(self):
+        if not self.is_number:
+            raise TypeError("can't truncate symbols and expressions")
+        elif self in (S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
```
I don't think this block is needed at all. This already raises TypeError.

abaokar-07 committed:
removed the code for complex args

abaokar-07 commented:
@asmeurer removed the code for complex, also the tests have passed

abaokar-07 commented:
@asmeurer are the lastest changes up to the mark?

abaokar-07 commented:
@asmeurer thanks for suggestions!
